api/src/main/scala/hmda/api/http/institutions/submissions/SubmissionBasePaths.scala

```diff
@@ -52,15 +53,19 @@ trait SubmissionBasePaths
             f <- fFilingsActor
             s <- fSubmissionsActor
             d <- (f ? GetFilingByPeriod(period)).mapTo[Filing]
-          } yield (s, d)
+          } yield (f, s, d)
 
           onComplete(fFiling) {
-            case Success((submissionsActor, filing)) =>
+            case Success((filingActor, submissionsActor, filing)) =>
               if (filing.period == period) {
                 submissionsActor ! CreateSubmission
                 val fLatest = (submissionsActor ? GetLatestSubmission).mapTo[Submission]
                 onComplete(fLatest) {
                   case Success(submission) =>
+                    if (filing.status != NotStarted) {
+                      val result = Await.result(filingActor ? UpdateFilingStatus(filing.copy(status = NotStarted)), 5.seconds)
```

Comment:
`Await` is a blocking operation, it should **never** be used in production code unless it is strictly necessary. It looks like a `for` comprehension could help here